### PR TITLE
Support customize dtype for dummy generators

### PR DESCRIPTION
**Pull request by JingyaHuang**
# What does this PR do?

* Support custom float dtype and int dtype when generating dummy inputs. (This will benefit sub-packages for generating specific dtype IO tensors before tracing).
* Will refractor the exporters in main repo as well, which currently cast from fp32 to other dtypes which is redundant.

### CODE REVIEW

File: optimum/commands/export/onnx.py [hunk 1]
```diff
@@ -61,6 +61,20 @@ def parse_args_onnx(parser):
...
+        choices=["int64", "int32", "int8"],
+        help="The data type of integer tensors, could be int64, int32, int8, default to int64.",
+    )
+    optional_group.add_argument(
+        "--float_dtype",
+        type=str,
+        default="fp32",
+        choices=["fp32", "fp16"],
+        help="The data type of float tensors, could be fp32, fp16, default to fp32.",
+    )
```
[the remark this answers is not in the document]
Comment (JingyaHuang) [reply optimum/commands/export/onnx.py:77]: Yeah, for the onnx case, let's remove int dtype and float dtype while keeping fp16 then.

[the remark this answers is not in the document]
Comment (JingyaHuang) [reply optimum/commands/export/onnx.py:77]: btw, the int dtype will only impact input dummy int tensors(default to int64 so far), won't cast float tensors

File: optimum/utils/input_generators.py [hunk 2]
```diff
@@ -115,20 +161,24 @@ def random_int_tensor(shape: List[int], max_value: int, min_value: int = 0, fram
                 The minimum value allowed.
             framework (`str`, *optional*, defaults to `"pt"`):
                 The requested framework.
+            dtype (`str`, *optional*, defaults to `"int64"`):
+                The dtype of the generated integer tensor. Could be "int64", "int32", "int8".
```
[the remark this answers is not in the document]
Comment (JingyaHuang) [reply optimum/utils/input_generators.py:165]: Yeah, but then I need to change all above...

[the remark this answers is not in the document]
Comment (JingyaHuang) [reply optimum/utils/input_generators.py]: Done

**Review by fxmarty — APPROVED** [review]

File: optimum/exporters/onnx/__main__.py [hunk 3]
```diff
@@ -264,8 +270,11 @@ def main_export(
 
     if fp16 is True and device == "cpu":
         raise ValueError(
-            "The --fp16 option is supported only when exporting on GPU. Please pass the option `--device cuda`."
+            "FP16 export is supported only when exporting on GPU. Please pass the option `--device cuda`."
         )
+        float_dtype = "fp16"
+    else:
+        float_dtype = "fp32"
```
Comment (fxmarty) [inline optimum/exporters/onnx/__main__.py:277]: actually breaking

**Pull request merged**